File: AppetiteAPI/AppetiteAPI/Models/Order.cs
using System;
using System.Collections.Generic;

namespace AppetiteAPI.Models
{
    public class Order
    {
        public int Id { get; set; }
        public User User { get; set; }
        public Restaurant Restaurant { get; set; }
        public double DeliveryCost { get; set; }
        public DateTime OrderReceivedTime { get; set; }
        public List<Product> Products { get; set; }
        public bool IsDone { get; set; }
    }
}

File: AngularClient/appetite/src/app/model/orderProcess/order.ts
import { IProduct } from "./product";
import { IRestaurant } from "./restaurant";

export interface IOrderAmount {
    name: string;
    amount: number;
}

export interface IOrderProductAPI {
    name: string
}

export interface IBodyOrder {
    "userEmail": string,
    "restaurantEmail": string,
    "products": IOrderProductAPI[],
}

export interface IOrder {
    "id": number,
    "restaurant": IRestaurant,
    "deliveryCost": number,
    "orderReceivedTime": string,
    "products": IProduct[],
    "isDone": boolean,
}
